src/components/cc-ssh-key-list/cc-ssh-key-list.js

```diff
@@ -109,29 +109,37 @@ export class CcSshKeyList extends LitElement {
     const name = rawName.trim();
     const publicKey = rawPublicKey.trim();
 
+    // looking for potential form errors
+    const isNameFilled = name?.length > 0;
+    const isPublicKeyFilled = publicKey?.length > 0;
+    const isPublicKeyPrivate = this._isPublicKeyPrivate(publicKey);
+
+    // updating the model
     this.createSshKeyForm = {
       state: 'idle',
       name: {
         value: name,
-        error: (name?.length > 0) ? null : 'required',
+        error: isNameFilled ? null : 'required',
       },
       publicKey: {
         value: publicKey,
-        error: (publicKey?.length > 0) ? null : 'required',
+        error: isPublicKeyFilled ? null : 'required',
       },
     };
 
-    // testing if submitted key is possibly a private key
-    if (publicKey.toLowerCase().match(' private ')) {
+    if (isPublicKeyPrivate) {
       this.createSshKeyForm.publicKey.error = 'private-key';
     }
```

Comment:
Is it a problem to trigger Lit's reactive thing with an assignement like `this.createSshKeyForm = {` and then modify the object?

Reply:
You're right, theoretically, we shouldn't do that. But I'm not sure Lit has 'the time' to react before the object is modified.